src/main/java/sirius/web/controller/Page.java

```diff
@@ -74,6 +76,30 @@
         return this;
     }
 
+    /**
+     * Easy method for supplying items the page contains to avoid boiler code.
+     * <p>
+     * The supplier provides the {@link Limit} of the current page for easy iteration
+     * over a result set.
+     * <p>
+     * This method should be supplied with a {@link List} which size was determined by
+     * the supplied {@link Limit} as this method does also is able to determine whether
+     * this page {@link Page#withHasMore(boolean) has more items to show} which are currently
+     * not being displayed.
+     *
+     * @param itemsSupplier the supplier to supply items to the current page limited by the provided limit
+     * @return the page itself for fluent method calls
+     */
+    public Page<E> withLimitedItemsSupplier(Function<Limit, List<E>> itemsSupplier) {
+        Limit supplierLimit = getCurrentLimit();
```

Comment:
can you set totalitems = supplieditems.size

Reply:
Not needed. The list is now sliced to the set page size

Reply:
not needed but pretty much expected to fill such fields in boileplate methode? wouldn't it make sense if i call page.getTotalItems to get the total number of items instead of 0 ?

Reply:
The number of total items is not known in this case. If the Limit is used with Limit::asPredicate a List is generated containing only pageSize + 1 elements.

Reply:
Then you have pagesize + 1 total items and not 0 total items